examples/Locomotion/Teleportation_Colliders/main.lua

```diff
@@ -0,0 +1,166 @@
+      motion.pose:rotate(angle, 0, 1, 0)
+      motion.thumbstickCooldown = motion.thumbstickCooldownTime
+    end
+  end
+  motion.blinkStopwatch = motion.blinkStopwatch + dt
+  motion.thumbstickCooldown = motion.thumbstickCooldown - dt
+end
+
+function motion.drawTeleport()
+  if lovr.headset.isTracked('right') then
+    -- Teleport target and curve
+    lovr.graphics.setColor(1, 1, 1, 0.1)
+    if motion.teleportValid then
+      lovr.graphics.setColor(1, 1, 0)
+      lovr.graphics.cylinder(motion.targetPosition, 0.05, math.pi/2,  1,0,0,  0.4, 0.4)
+      lovr.graphics.setColor(1, 1, 1)
+    end
+    lovr.graphics.setLineWidth(4)
+    lovr.graphics.line(motion.teleportCurve:render(30))
+  end
+  -- Teleport blink, modeled as gaussian function
+  local blinkAlpha = math.exp(-(motion.blinkStopwatch/ 0.25 / motion.blinkTime)^2)
+  lovr.graphics.setColor(0,0,0, blinkAlpha)
+  lovr.graphics.fill()
+end
+
+
+
+local function makeColumn(x, z, height, color)
+  height = math.abs(height)
+  local collider = physicsWorld:newCylinderCollider(x, height / 2, z, 2, height)
+  local shape = collider:getShapes()[1]
+  collider:setOrientation(math.pi/2,  1,0,0)
+  collider:setKinematic(true)
+  local column = {
+    collider=collider,
+    shape=shape,
+    color=color,
+  }
+  table.insert(columns, column)
+end
+
+function lovr.load()
+  physicsWorld = lovr.physics.newWorld()
+  -- Some scenery
+  lovr.math.setRandomSeed(0)
+  local goldenRatio = (math.sqrt(5) + 1) / 2
+  local goldenAngle = (2 - goldenRatio) * (2 * math.pi)
+  local k = 2.5
+  for i = 1, 100 do
+    local r = math.sqrt(i) * k
+    local x = math.cos(goldenAngle * i) * r
+    local z = math.sin(goldenAngle * i) * r
+    local height = 5 + 10 * math.exp(-(r/50)^2)
+    local shade = 0.1 + 0.3 * lovr.math.random()
+    local color = {shade, shade, shade}
+    makeColumn(x, z, height, color)
+    -- set initial VR position to the top of center column
+    if i == 1 then
+      motion.pose:set(x, height, z)
+    end
+  end
+end
+
+function lovr.update(dt)
+  motion.teleport(dt)
+end
+
+function lovr.draw()
+  lovr.graphics.setBackgroundColor(0.1, 0.1, 0.1)
+  lovr.graphics.transform(mat4(motion.pose):invert())
+  -- Render hands
+  lovr.graphics.setColor(1,1,1)
+  local radius = 0.04
+  for _, hand in ipairs(lovr.headset.getHands()) do
+    -- Whenever pose of hand or head is used, need to account for VR movement
+    local poseRW = mat4(lovr.headset.getPose(hand))
+    local poseVR = mat4(motion.pose):mul(poseRW)
+    poseVR:scale(radius)
+    lovr.graphics.sphere(poseVR)
+  end
+  -- Render columns
+  for i, column in ipairs(columns) do
+    local x,y,z, angle, ax,ay,az = column.collider:getPose()
+    local l, r = column.shape:getLength(), column.shape:getRadius()
+    lovr.graphics.setColor(unpack(column.color))
+    lovr.graphics.cylinder(x,y,z, l, angle, ax,ay,az, r, r, true, 20)
+  end
+  -- Teleportation curve and target, rendering of blinking overlay
+  motion.drawTeleport()
+end
```

Comment:
I added `lovr.graphics.setColor(1,1,1)` at the end of draw so the mirror window shows up on desktop

Reply:
I don't use mirror window so I don't know how it behaved before setColor was added. Is this something we should fix in mirror rendering code? I don't think it should dependent on color state left behind by draw function.

Reply:
It will be fixed soon https://github.com/bjornbytes/lovr/issues/313